src/types.rs

```diff
@@ -534,6 +604,24 @@ impl TxEventsResponse {
     }
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+pub struct SignedMsgResponse {
+    pub signed_msg_order_results: Vec<SignedMsgOrderResult>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct SignedMsgOrderResult {
+    pub hash: String,
+    pub status: String,
+    pub error: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum PlaceOrderResponse {
+    Tx(TxResponse),
+    SignedMsg(SignedMsgResponse),
+}
```

Comment:
should remove nesting
```suggestion
pub enum PlaceOrderResponse {
    #[serde(flatten)]
    Tx(TxResponse),
    #[serde(flatten)]
    SignedMsg(SignedMsgResponse),
}
```